include/FunctionInfosManager.h

```diff
@@ -3,6 +3,12 @@
 
 #include "FunctionInfo.h"
 
+// uses the AST of a file to get information about functions *not* from the point
+// of view of the LLVM IR, but as it is seen in the code. this is important
+// as the IR can desugar functions or perform optimizations which
+// change the arity of those functions; if the arity is different that what
+// the user actually sees, then we may (or may not) analyze resources
```

Comment:
I'm not sure what you mean by "may (or may not)" in this sentence. I understand how changing the arity of a function might complicate our analysis, but the wording here is too vague. Can you be more specific about this class' role?